user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
Should this be a [`401 NotAuthenticated`](http://tomchristie.github.io/rest-framework-2-docs/api-guide/exceptions#notauthenticated)?